lib/uniffi-zcash/src/zcash_primitives/sapling/keys/extended_full_viewing_key.rs

```diff
@@ -0,0 +1,88 @@
+use std::sync::Arc;
+
+use zcash_primitives::zip32::ExtendedFullViewingKey;
+
+use crate::{
+    ZcashChildIndex, ZcashDiversifiableFullViewingKey, ZcashDiversifierIndex,
+    ZcashDiversifierIndexAndPaymentAddress, ZcashError, ZcashPaymentAddress, ZcashResult,
+};
+
+impl From<ExtendedFullViewingKey> for ZcashExtendedFullViewingKey {
+    fn from(key: ExtendedFullViewingKey) -> Self {
+        ZcashExtendedFullViewingKey(key)
+    }
+}
+
+impl From<&ZcashExtendedFullViewingKey> for ExtendedFullViewingKey {
+    fn from(key: &ZcashExtendedFullViewingKey) -> Self {
+        key.0.clone()
+    }
+}
+
+pub struct ZcashExtendedFullViewingKey(ExtendedFullViewingKey);
+
+impl ZcashExtendedFullViewingKey {
+    pub fn from_bytes(bytes: Vec<u8>) -> ZcashResult<Self> {
```

Comment:
Nitpick: Could we use `[ByRef]` here ?

Do not take this into account if this was due to a code move. I think we need a general review in all code for this details :)